src/ovs/aca_arp_responder.cpp

```diff
@@ -0,0 +1,405 @@
+// Copyright 2019 The Alcor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "aca_arp_responder.h"
+#include "aca_log.h"
+#include "aca_ovs_l2_programmer.h"
+#include "aca_ovs_control.h"
+#include "aca_util.h"
+#include <shared_mutex>
+#include <arpa/inet.h>
+
+using namespace std;
+
+namespace aca_arp_responder
+{
+ACA_ARP_Responder::ACA_ARP_Responder(){
+  _init_arp_db();
+  _init_arp_ofp();
+}
+
+ACA_ARP_Responder::~ACA_ARP_Responder(){
+  _deinit_arp_db();
+  _deinit_arp_ofp();
+}
+
+void ACA_ARP_Responder::_init_arp_db(){
+  _arp_db.clear();
+}
+
+void ACA_ARP_Responder::_deinit_arp_db(){
+  _arp_db.clear();
+}
+
+void ACA_ARP_Responder::_init_arp_ofp(){
+  int overall_rc = EXIT_SUCCESS;
+  unsigned long not_care_culminative_time;
+
+  aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().execute_openflow_command(
+    "add-flow br-tun \"table=0,priority=50,arp,arp_op=1, actions=CONTROLLER\"",
+    not_care_culminative_time, overall_rc);
+  return;
+}
+
+void ACA_ARP_Responder::_deinit_arp_ofp(){
+  unsigned long not_care_culminative_time;
+  int overall_rc = EXIT_SUCCESS;
+
+  aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().execute_openflow_command(
+          "del-flows br-tun \"arp,arp_op=1\"",
+          not_care_culminative_time, overall_rc);
+  
+  aca_ovs_l2_programmer::ACA_OVS_L2_Programmer::get_instance().execute_openflow_command(
```

Comment:
Don't need to add back the old arp rule once we switch to the on-demand arp responder. BTW, we will need to remove the original arp rules in int ACA_OVS_L2_Programmer::setup_ovs_bridges_if_need().